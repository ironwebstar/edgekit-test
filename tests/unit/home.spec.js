import { shallowMount } from "@vue/test-utils";
import LoginPage from "@/views/Home.vue";

describe("HelloWorld.vue", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(LoginPage);
  });

  // snapshot test
  it("should render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("call login action when trigger login button.", () => {
    const vm = wrapper.vm;
    const loginMock = jest.fn();
    const validEmailMock = jest.fn();
    vm.checkForm = loginMock;
    vm.validEmail = validEmailMock;

    wrapper.find("button").trigger("click");
    expect(loginMock).toHaveBeenCalled();
    expect(validEmailMock).not.toHaveBeenCalled();
  });
});
