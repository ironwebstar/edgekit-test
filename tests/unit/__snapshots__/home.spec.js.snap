// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue should render correctly 1`] = `
<div class="login flex column">
  <p class="text-h2--max">Log in</p>
  <p class="text-h2--max">with EdgeKit</p>
  <div class="flex row email input-wrapper"><input class="input" placeholder="Work email">
    <!--v-if-->
  </div>
  <div class="flex row password input-wrapper"><input class="input" placeholder="Password">
    <div class="input-context">
      <!--v-if-->
      <div class="text-primary">Forgot password?</div>
    </div>
  </div>
  <div class="flex row password input-wrapper"><button type="submit" class="btn btn-login"> Log in </button></div>
</div>
`;
